linera-core/src/chain_worker/actor.rs

```diff
@@ -312,9 +318,11 @@ where
         }
 
         drop(self.worker);
-        self.service_runtime_thread
-            .await
-            .expect("Service runtime thread should not panic");
+        if let Some(thread) = self.service_runtime_thread {
```

Comment:
Optional: IIRC, you can also include the `drop` above inside the `if` body. The goal of it is to ensure that worker is dropped and releases the endpoint to the thread, so that the thread knows it should stop.